05week/checkers.js

```diff
@@ -52,16 +68,62 @@ function Board() {
     console.log(string);
   };
 
+  // populates the grid with checkers
+ this.createCheckers = () => {
+   // loops through the 8 rows
+   for (let row = 0; row < 8; row++) {
+     // ignores rows which should be empty
+     if (row === 3 || row === 4) continue;
+     // loops through the 8 columns
+     for (let col = 0; col < 8; col++) {
+       // sets current color based on the current row
+       let color = (row < 3 ? 'white' : 'black');
+       // alternates cells to populate with either white or black checkers
+       // then pushes checker to array named checkers
+       if (row % 2 === 0 && col % 2 === 1) {
+         this.grid[row][col] = new Checker(color);
+         this.checkers.push(this.grid[row][col]);
+       } else if (row % 2 === 1 && col % 2 === 0) {
+         this.grid[row][col] = new Checker(color);
+         this.checkers.push(this.grid[row][col]);
+       }
+     }
+   }
+};
+
+
   // Your code here
-}
-function Game() {
 
-  this.board = new Board();
 
-  this.start = function() {
+Class Game() {
+  //Create a new board
+  //start on the black checker's turns
+  //populate the board and checkers on it
+  this.board = new Board();
+  this.turn = 'black';
+  this.start = () => {
     this.board.createGrid();
-    // Your code here
-  };
+    this.board.createCheckers();
+  }
+  //moveChecker(), move a checker from the beginning location to the end location
+  //split the player input into an array.  The first number is the Y axis and the second number is the X axis
+  //Move from startingPositionto endingPosition
+  this.moveChecker = (startingPosition, endingPosition) => {
+    const beginning = startingPosition.split('');
+    const end = endingPosition.split('');
```

Comment:
These are unnecessary. You can access characters in a string with brackets and their index, no need to make into an array.